pandas/tests/series/methods/test_shift.py

```diff
@@ -276,3 +276,20 @@ def test_shift_dt64values_int_fill_deprecated(self):
 
         expected = pd.Series([pd.Timestamp(0), ser[0]])
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("periods", [1, 2, 3, 4])
+    def test_shift_preserve_freqstr(self, periods):
+        # GH#21275
+        data = list(range(0, periods))
```

Comment:
You can pass a range directly to the DataFrame constructor so could just be range(periods)